Sources/Sentry/SentryUIApplication.m

```diff
@@ -56,6 +56,105 @@ - (UIApplication *)sharedApplication
     return result;
 }
 
+- (NSArray<UIViewController *> *)relevantViewControllers
+{
+    NSMutableArray *result = [NSMutableArray array];
+
+    NSArray<UIWindow *> *windows = [self windows];
+    if ([windows count] == 0)
+        return nil;
+
+    for (UIWindow *window in windows) {
+        NSArray<UIViewController *> *vcs = [self relevantViewControllerFromWindow:window];
+        if (vcs != nil) {
+            [result addObjectsFromArray:vcs];
+        }
+    }
+
+    return result;
+}
+
+- (NSArray<UIViewController *> *)relevantViewControllerFromWindow:(UIWindow *)window
+{
+    UIViewController *rootViewController = window.rootViewController;
+    if (rootViewController == nil) {
+        return nil;
+    }
+
+    NSMutableArray<UIViewController *> *result = @[ rootViewController ].mutableCopy;
+    NSUInteger index = 0;
+
+    while (index < result.count) {
+        UIViewController *topVC = result[index];
+        // If the view controller is presenting another one, usually in a modal form.
+        if (topVC.presentedViewController != nil) {
+            [result replaceObjectAtIndex:index withObject:topVC.presentedViewController];
```

Comment:
Two questions:
1. Why replace instead of reporting them both? The combination might be interesting information.
2. Even if we replace presenters with modals, do we also want to do that for UIAlertControllers? (dialogs and share sheets are both presented via that API which is also a UIVC subclass)

Reply:
I guess breadcrumbs could be used to reconstruct the trail of presenters and other navigation sequences that led to the topmost vc(s) currently visible.